src/UnitTests/GitHub.App/Models/RepositoryHostTests.cs

```diff
@@ -161,6 +161,66 @@ public class TheLoginMethod : TestBaseClass
             Assert.Equal("jiminy", loginInfo.UserName);
             Assert.Equal("T0k3n", loginInfo.Password);
         }
+
+        [Fact]
+        public async Task SupportsGistIsTrueWhenGistScopeIsPresent()
+        {
+            var apiClient = Substitute.For<IApiClient>();
+            apiClient.HostAddress.Returns(HostAddress.GitHubDotComHostAddress);
+            apiClient.GetUser().Returns(Observable.Return(CreateUserAndScopes("baymax", new[] { "gist" })));
+            var hostCache = new InMemoryBlobCache();
+            var modelService = new ModelService(apiClient, hostCache, Substitute.For<IAvatarProvider>());
+            var loginCache = new TestLoginCache();
+            var host = new RepositoryHost(apiClient, modelService, loginCache, Substitute.For<ITwoFactorChallengeHandler>());
+
+            var result = await host.LogIn("baymax", "aPassword");
+
+            Assert.Equal(AuthenticationResult.Success, result);
+            Assert.True(host.SupportsGist);
+        }
+
+        [Fact]
+        public async Task SupportsGistIsFalseWhenGistScopeIsNotPresent()
+        {
+            var apiClient = Substitute.For<IApiClient>();
+            apiClient.HostAddress.Returns(HostAddress.GitHubDotComHostAddress);
+            apiClient.GetUser().Returns(Observable.Return(CreateUserAndScopes("baymax", new[] { "foo" })));
+            var hostCache = new InMemoryBlobCache();
+            var modelService = new ModelService(apiClient, hostCache, Substitute.For<IAvatarProvider>());
+            var loginCache = new TestLoginCache();
+            var host = new RepositoryHost(apiClient, modelService, loginCache, Substitute.For<ITwoFactorChallengeHandler>());
+
+            var result = await host.LogIn("baymax", "aPassword");
+
+            Assert.Equal(AuthenticationResult.Success, result);
+            Assert.False(host.SupportsGist);
+        }
+
+        [Fact]
+        public async Task SupportsGistIsTrueWhenScopesAreNull()
+        {
+            // TODO: Check assumptions here. From my conversation with @shana it seems that the first login
+            // will be done with basic auth and from then on a token will be used. So if it's the first login,
+            // it's from this version and so gists will be supported. However I've been unable to repro this
+            // behavior.
```

Comment:
Yeah, I added this comment with this commit because I wasn't able to repro the case where a response comes back without the OAuth headers, though I'm pretty certain I've seen it in the past, because that's why I introduced the complexity a while back where I checked whether we're using basic auth.